internal/processing/istio/request_authentication_processor.go

```diff
@@ -0,0 +1,156 @@
+package istio
+
+import (
+	"context"
+	"fmt"
+
+	gatewayv1beta1 "github.com/kyma-incubator/api-gateway/api/v1beta1"
+	"github.com/kyma-incubator/api-gateway/internal/builders"
+	"github.com/kyma-incubator/api-gateway/internal/processing"
+	"istio.io/api/security/v1beta1"
+	securityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// RequestAuthenticationProcessor is the generic processor that handles the Istio Request Authentications in the reconciliation of API Rule.
+type RequestAuthenticationProcessor struct {
+	Creator RequestAuthenticationCreator
+}
+
+// RequestAuthenticationCreator provides the creation of Request Authentications using the configuration in the given APIRule.
+// The key of the map is expected to be unique and comparable with the
+type RequestAuthenticationCreator interface {
```

Comment:
This interface seems not to be needed, as in all likelihood
 there will be only one implementation for Istio handler